bezier-rs/lib/src/lib.rs

```diff
@@ -522,6 +571,8 @@ impl Bezier {
 		self.apply_transformation(&|point| point + translation)
 	}
 
+	// TODO: Change this to `intersect(&self, other: &Bezier)` to also work on quadratic and cubic segments
+	// TODO: (or keep this and add two more functions that perform the logic, and make the `intersect` function call the correct one)
 	/// Returns a list of points where the provided line segment intersects with the Bezier curve.
 	/// - `line` - A line segment expected to be received in the format of `[start_point, end_point]`.
```

Comment:
Something should be written about the case where a segment is colinear with the line and thus has infinite intersection points. (I'd expect there should be zero in that case, if floating point imprecision doesn't make that too difficult.)

Reply:
I added a test to ensure that `solve_linear` returns 0 when the line is on the x-axis.